lib/directory-view.coffee

```diff
@@ -44,8 +44,9 @@ class DirectoryView extends HTMLElement
     @expand() if @directory.isExpanded
 
   updateStatus: ->
-    @classList.remove('status-ignored', 'status-modified', 'status-added')
-    @classList.add("status-#{@directory.status}") if @directory.status?
+    if atom.config.get('tree-view.enableVcsColoring')
+      @classList.remove('status-ignored', 'status-modified', 'status-added')
+      @classList.add("status-#{@directory.status}") if @directory.status?
```

Comment:
Should this be:

```coffee
@classList.remove('status-ignored', 'status-modified', 'status-added')
@classList.add("status-#{@directory.status}") if atom.config.get('tree-view.enableVcsColoring') and @directory.status?
```

This way the status classes are always removed, just selectively added. I can imagine scenarios where Atom is started with the setting enabled and then it is disabled without restarting Atom or reloading the window ... the classes will still be there for some parts of the tree, but not others. And I don't see any tests covering this particular scenario.